lib/clouds/shared_functions.py

```diff
@@ -230,22 +230,22 @@ def wait_for_instance_ready(self, obj_attr_list, time_mark_prs) :
             if _pooling_time <= _wait :
                 _actual_wait = _wait - _pooling_time
             else :
-                _msg = "The time spent on pooling for \"ready\" status (" + str(_pooling_time) 
+                _msg = "The time spent on pooling for \"ready\" status for "
+                _msg += obj_attr_list["log_string"] + "(" + str(_pooling_time) 
                 _msg += " s) is actually longer than the "
                 _msg += "interval between pooling attempts (" + str(_wait) + " s)."
                 cbdebug(_msg, True)
                 _actual_wait = 0
 
-            # There is still some reconciliation to be done here. If vpn_only is used, then only openvpn-initiated callbacks should set the pending attribute, not userdata scripts. There is a distinction. It also means that public_cloud_ip should be set as well and that access to pending attributes is a requirement. See get_ip_address from do_cloud_ops.py
-            # Also "use_vpn_ip" is used throughout the scripts and codebase, so use_vpn_ip is already reserved to ensure that vpn_only works as it did before.
-            # So any changes to use_vpn_ip need to be conditionalized with an extra check to vpn_only as well.
-            if str(obj_attr_list["use_vpn_ip"]).lower() != "false" and str(obj_attr_list["vpn_only"]).lower() == "false" :
+            if str(obj_attr_list["use_vpn_ip"]).lower() != "false" :
```

Comment:
Is "VPN_ONLY" deprecated? or implied now? How does that work?

Reply:
No, not deprecated at all! VPN_ONLY is used when deciding the IP address of the Object Store and the File Store. Please look at the `create_bootstrap_script` method in "shared_function.py"

Reply:
I'll need to test this first using a separate PR.